src/versionbits.cpp

```diff
@@ -105,6 +104,36 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
     return state;
 }
 
+// return the numerical statistics of blocks signalling the specified BIP9 condition in this current period
+BIP9Stats AbstractThresholdConditionChecker::GetStateStatisticsFor(const CBlockIndex* pindex, const Consensus::Params& params) const
+{
+    BIP9Stats stats;
+    
+    stats.period = Period(params);
+    stats.threshold = Threshold(params);
+    
+    if (pindex == NULL)
+        return stats;
+
+    // Find beginning of period
+    const CBlockIndex* pindexFirstInPeriod = pindex->GetAncestor(pindex->nHeight - ((pindex->nHeight + 1) % stats.period));
```

Comment:
Ah I think you're right. I copied this line from `AbstractThresholdConditionChecker::GetStateFor` 

https://github.com/pinheadmz/bitcoin/blob/83538dd86dc426dc252714d4ca1ae91ad2c5ac24/src/versionbits.cpp#L30-L33

We'll call it `pindexEndOfPrevPeriod` ?